src/coreclr/jit/targetriscv64.cpp

```diff
@@ -24,4 +24,154 @@ const regNumber fltArgRegs [] = {REG_FLTARG_0, REG_FLTARG_1, REG_FLTARG_2, REG_F
 // clang-format on
 
+//-----------------------------------------------------------------------------
+// RiscV64Classifier:
+//   Construct a new instance of the RISC-V 64 ABI classifier.
+//
+// Parameters:
+//   info - Info about the method being classified.
+//
+RiscV64Classifier::RiscV64Classifier(const ClassifierInfo& info)
+    : m_info(info)
+    , m_intRegs(intArgRegs, ArrLen(intArgRegs))
+    , m_floatRegs(fltArgRegs, ArrLen(fltArgRegs))
+{
+}
+
+//-----------------------------------------------------------------------------
+// Classify:
+//   Classify a parameter for the RISC-V 64 ABI.
+//
+// Parameters:
+//   comp           - Compiler instance
+//   type           - The type of the parameter
+//   structLayout   - The layout of the struct. Expected to be non-null if
+//                    varTypeIsStruct(type) is true.
+//   wellKnownParam - Well known type of the parameter (if it may affect its ABI classification)
+//
+// Returns:
+//   Classification information for the parameter.
+//
+ABIPassingInformation RiscV64Classifier::Classify(Compiler*    comp,
+                                                  var_types    type,
+                                                  ClassLayout* structLayout,
+                                                  WellKnownArg /*wellKnownParam*/)
+{
+    assert(!m_info.IsVarArgs); // TODO: varargs currently not supported on RISC-V
+
+    StructFloatFieldInfoFlags flags     = STRUCT_NO_FLOAT_FIELD;
+    unsigned                  intFields = 0, floatFields = 0;
+    unsigned                  passedSize;
+
+    if (varTypeIsStruct(type))
+    {
+        passedSize = structLayout->GetSize();
+        if (passedSize > MAX_PASS_MULTIREG_BYTES)
+        {
+            passedSize = TARGET_POINTER_SIZE; // pass by reference
+        }
+        else if (!structLayout->IsBlockLayout())
+        {
+            flags = (StructFloatFieldInfoFlags)comp->info.compCompHnd->getRISCV64PassStructInRegisterFlags(
+                structLayout->GetClassHandle());
+
+            if ((flags & STRUCT_FLOAT_FIELD_ONLY_ONE) != 0)
+            {
+                floatFields = 1;
+            }
+            else if ((flags & STRUCT_FLOAT_FIELD_ONLY_TWO) != 0)
+            {
+                floatFields = 2;
+            }
+            else if (flags != STRUCT_NO_FLOAT_FIELD)
+            {
+                assert((flags & (STRUCT_FLOAT_FIELD_FIRST | STRUCT_FLOAT_FIELD_SECOND)) != 0);
+                floatFields = 1;
+                intFields   = 1;
+            }
+        }
+    }
+    else
+    {
+        assert(genTypeSize(type) <= TARGET_POINTER_SIZE);
+
+        if (varTypeIsFloating(type))
+            floatFields = 1;
+
+        passedSize = genTypeSize(type);
+    }
+
+    assert((floatFields > 0) || (intFields == 0));
+
+    auto PassSlot = [this](bool inFloatReg, unsigned offset, unsigned size) -> ABIPassingSegment {
```

Comment:
Nit: Most places within the JIT that use lambdas use camelCase naming convention.

No need to address this here.